src/TextField/TextField.jsx

```diff
@@ -289,12 +279,15 @@ const TextField = React.createClass({
     ) : null;
 
     let floatingLabelTextElement = floatingLabelText ? (
-      <label
-        style={this.prepareStyles(styles.floatingLabel, this.props.floatingLabelStyle)}
+      <TextFieldLabel
+        muiTheme={this.state.muiTheme}
+        style={this.mergeStyles(styles.floatingLabel, this.props.floatingLabelStyle)}
```

Comment:
My initial thoughts, but it's not that simple. Doing this will couple the components a LOT. In order to calculate the color, `hasError`, `hasValue` and `errorColor` must too be passed onwn only to calculate a simple `color` so I left that calculation to the TextField.